spec/models/acts_as_customizable/calculated_value_spec.rb

```diff
@@ -381,4 +382,194 @@ def custom_field_values(*) = nil
       end
     end
   end
+
+  describe "#calculate_custom_fields error handling" do
+    def expect_calculated_value_error(calc_val_cf, project, error_code, message_part = nil)
+      error = calc_val_cf.first_calculation_error(project)
+      expect(error&.error_code).to eq(error_code)
+
+      if message_part
+        expect(calculated_value_error_msg(error)).to include(message_part)
+      end
+    end
+
+    let(:project) { create(:project) }
+    let(:enabled_custom_fields) { {} }
+
+    before do
+      enabled_custom_fields.each do |cf, value|
+        create(:custom_value, customized: project, custom_field: cf, value:)
+      end
+    end
+
+    describe "division by zero" do
+      let(:cf_div) { create(:calculated_value_project_custom_field, projects: [project], formula: "5 / 0") }
+      let(:cf_add) { create(:calculated_value_project_custom_field, projects: [project], formula: "5 + 0") }
+
+      let(:enabled_custom_fields) do
+        { cf_div => nil, cf_add => nil }
+      end
+
+      it "creates a mathematical error for division by zero" do
+        project.calculate_custom_fields([cf_div, cf_add])
+
+        expect(cf_add.calculated_value_errors.where(customized: project)).to be_empty
+
+        errors = cf_div.calculated_value_errors.where(customized: project)
+        expect(errors.size).to eq(1)
+        expect(errors.first&.error_code).to eq("ERROR_MATHEMATICAL")
+      end
+    end
+
+    describe "missing value" do
+      let(:cf_int) { create(:integer_project_custom_field, projects: [project]) }
+      let(:cv1) do
+        create(:calculated_value_project_custom_field, :skip_validations, projects: [project], formula: "5 + #{cf_int}")
+      end
+      let(:cv2) { create(:calculated_value_project_custom_field, :skip_validations, projects: [project], formula: "5 + #{cv1}") }
+      let(:cv3) { create(:calculated_value_project_custom_field, :skip_validations, projects: [project], formula: "5 + 5") }
+
+      let(:enabled_custom_fields) do
+        { cv1 => nil, cv2 => nil, cf_int => nil, cv3 => nil }
+      end
+
+      it "creates a missing value error" do
+        project.calculate_custom_fields([cv1, cv2, cv3])
+
+        expect(cv3.calculated_value_errors.where(customized: project)).to be_empty
+
+        expect_calculated_value_error(cv1, project, "ERROR_MISSING_VALUE", cf_int.name)
+        expect_calculated_value_error(cv2, project, "ERROR_MISSING_VALUE", cv1.name)
+      end
+    end
+
+    describe "disabled value" do
+      let(:cf_int) { create(:integer_project_custom_field) }
```

Comment:
I think better to explicitly show that it is not enabled:
```suggestion
      let(:cf_int) { create(:integer_project_custom_field, projects: []) }
```